internal/planner/server.go

```diff
@@ -0,0 +1,136 @@
+/*
+ * Copyright 2025 Canonical Ltd.
+ * See LICENSE file for licensing details.
+ *
+ * Package planner provides HTTP handlers for the Planner API service.
+ */
+
+package planner
+
+import (
+	"context"
+	"encoding/hex"
+	"encoding/json"
+	"errors"
+	"fmt"
+	"net/http"
+	"strings"
+
+	"github.com/canonical/github-runner-operators/internal/database"
+	"go.opentelemetry.io/contrib/instrumentation/net/http/otelhttp"
+)
+
+type contextKey string
+
+const tokenNameKey contextKey = "tokenName"
+const createFlavorPattern = "/api/v1/flavors/{name}"
+
+// FlavorStore is a small interface that matches the relevant method on internal/database.Database.
+type FlavorStore interface {
+	AddFlavor(ctx context.Context, flavor *database.Flavor) error
+	ListFlavors(ctx context.Context, platform string) ([]database.Flavor, error)
+	GetPressures(ctx context.Context, platform string, flavors ...string) (map[string]int, error)
+	VerifyAuthToken(ctx context.Context, token [32]byte) (string, error)
+}
+
+// Server holds dependencies for the planner HTTP handlers.
+type Server struct {
+	metrics *Metrics
+	mux     *http.ServeMux
+	store   FlavorStore
+}
+
+// NewServer creates a new Server with the given dependencies.
+func NewServer(store FlavorStore, metrics *Metrics) *Server {
+	s := &Server{store: store, mux: http.NewServeMux(), metrics: metrics}
+
+	// Register routes
+	s.mux.Handle("POST "+createFlavorPattern, otelhttp.WithRouteTag(createFlavorPattern, s.AuthMiddleware(http.HandlerFunc(s.createFlavor))))
+
+	return s
+}
+
+// ServeHTTP routes incoming HTTP requests to the appropriate handler methods.
+func (s *Server) ServeHTTP(w http.ResponseWriter, r *http.Request) {
+	s.mux.ServeHTTP(w, r)
+}
+
+// AuthMiddleware is an HTTP middleware that checks for a valid user token
+// in the Authorization header.
+func (s *Server) AuthMiddleware(next http.Handler) http.Handler {
```

Comment:
I think we agreed to not add this initially (we have a dedicated user story for authentication: https://warthogs.atlassian.net/browse/ISD-4479). May I ask how we add the auth token to the database? Would this need a manual database operation?

Reply:
ok i will delete this scope! for adding auth token i use this operation inside the integration test
```
	// Create an auth token to authenticate requests
	tokenName := "it-planner-" + time.Now().Format("20060102-150405.000000000")
	token, err := db.CreateAuthToken(ctx, tokenName)
	if err != nil {
		t.Fatalf("create auth token failed: %v", err)
	}
	tokenHex := hex.EncodeToString(token[:])
```